### PR TITLE
Review and some changes

### PR DESCRIPTION
**Pull request by sujeebant**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -73,59 +84,73 @@ class Masq {
...
   get (key) {
-    return new Promise((resolve, reject) => {
-      if (!this.profile) return reject(Error('No profile selected'))
-      const db = this.dbs[this.profile]
-      db.get(key, (err, nodes) => {
-        if (err) return reject(err)
-        if (!nodes.length) return resolve(nodes[0])
-        resolve(nodes[0].value)
+    try {
+      let db = this._getDB()
```
Comment (G-Ray) [inline src/index.js]: you should not need to wrap this with try/catch, if you move `_getDB()` inside the promise and it throws, then `get()` will reject the error too (

File: src/index.js [hunk 2]
```diff
@@ -73,59 +84,73 @@ class Masq {
...
+    }
   }
 
-  /**
-   * If this is the first time, this.dbs.profiles is empty.
-   * We need to get masq-profiles hyperdb key of masq.
-   */
-
-  requestMasqAccess () {
+  _initSwarmWithDataHandler (dataHandler) {
```
Comment (G-Ray) [inline src/index.js]: Nice

File: src/index.js [hunk 3]
```diff
@@ -51,15 +55,22 @@ class Masq {
...
+        return resolve(Promise.all(promiseArr))
+      })
+    })
+  }
+
+  getProfileByID (id) {
+    return new Promise((resolve, reject) => {
+      this.dbs.profiles.get(`/profiles/${id}`, (err, nodes) => {
+        if (err) return reject(err)
+        if (!nodes || !nodes[0] || !nodes[0].value) return reject(Error('Errooooor : ' + id + ' ; ' + JSON.stringify(nodes)))
```
[the remark this answers is not in the document]
Comment (sujeebant) [reply src/index.js]: fixed !

File: package.json [hunk 4]
```diff
@@ -7,6 +7,11 @@
     "test": "eslint src/ test/ && jest",
     "build": "webpack -p"
   },
+  "jest": {
```
[the remark this answers is not in the document]
Comment (sujeebant) [reply package.json]: it isn't, removed it !

File: src/index.js (hunk 1 above)
Comment (G-Ray) [reply src/index.js]: https://mrcoles.com/javascript-promises-and-errors/

Comment (sujeebant) [reply src/index.js]: fixed

**Pull request merged**